kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/CoordinatorThreadFactoryImpl.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.tabular.iceberg.connect.channel;
+
+import io.tabular.iceberg.connect.IcebergSinkConfig;
+import io.tabular.iceberg.connect.data.Utilities;
+import java.util.Collection;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.ConsumerGroupDescription;
+import org.apache.kafka.clients.admin.MemberDescription;
+import org.apache.kafka.common.ConsumerGroupState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class CoordinatorThreadFactoryImpl implements CoordinatorThreadFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(CoordinatorThreadFactoryImpl.class);
+  private final AdminFactory adminFactory;
+  private final ConsumerFactory consumerFactory;
+  private final ProducerFactory producerFactory;
+
+  CoordinatorThreadFactoryImpl(
+      AdminFactory adminFactory, ConsumerFactory consumerFactory, ProducerFactory producerFactory) {
+    this.adminFactory = adminFactory;
+    this.consumerFactory = consumerFactory;
+    this.producerFactory = producerFactory;
+  }
+
+  @Override
+  public CoordinatorThread create(IcebergSinkConfig config) {
+    final Collection<MemberDescription> members;
+    try (Admin admin = adminFactory.create(config.kafkaProps())) {
+      ConsumerGroupDescription groupDesc =
+          KafkaUtils.consumerGroupDescription(config.connectGroupId(), admin);
+      if (groupDesc.state() == ConsumerGroupState.STABLE) {
+        members = groupDesc.members();
+      } else {
+        throw new IllegalStateException("Consumer group is not stable, cannot start coordinator");
+      }
```

Comment:
Another change: Previously when the group wasn't stable we simply wouldn't create/start a CoordinatorThread. If a group isn't stable, it's likely the group will be rebalanced soon so it makes sense. But I think we should just throw an exception and get on with the rebalance in that case. 

I can go either way I suppose.

Reply:
Won't throwing the exception cause further rebalances?